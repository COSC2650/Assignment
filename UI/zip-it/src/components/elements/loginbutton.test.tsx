import { configure, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
//import { expect } from '@jest/globals';
import { Button } from '@chakra-ui/react';
import { IconButton} from '@chakra-ui/react';

jest.useFakeTimers();

configure({ adapter: new Adapter() });

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect,
}));

describe('LoginButton', () => {
  it('Should render correctly', () => {
    

    const component = render(
      <>
        <IconButton aria-label="Log in" />
        <Button />
      </>
    );
    expect(component).toMatchSnapshot();
  });
});
