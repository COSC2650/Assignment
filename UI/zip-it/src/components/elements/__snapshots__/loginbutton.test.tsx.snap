// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton Should render correctly 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "aria-label": "Log in",
      "class": "chakra-button css-198bfjk",
      "type": "button",
    },
    "children": Array [],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Node {
      "attribs": Object {
        "class": "chakra-button css-5c2176",
        "type": "button",
      },
      "children": Array [],
      "name": "button",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": [Circular],
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
        "type": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
        "type": undefined,
      },
    },
    "parent": Node {
      "children": Array [
        [Circular],
        Node {
          "attribs": Object {
            "class": "chakra-button css-5c2176",
            "type": "button",
          },
          "children": Array [],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "type": undefined,
          },
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-label": undefined,
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-label": undefined,
      "class": undefined,
      "type": undefined,
    },
  },
  "1": Node {
    "attribs": Object {
      "class": "chakra-button css-5c2176",
      "type": "button",
    },
    "children": Array [],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        Node {
          "attribs": Object {
            "aria-label": "Log in",
            "class": "chakra-button css-198bfjk",
            "type": "button",
          },
          "children": Array [],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-label": undefined,
            "class": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-label": undefined,
            "class": undefined,
            "type": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": Node {
      "attribs": Object {
        "aria-label": "Log in",
        "class": "chakra-button css-198bfjk",
        "type": "button",
      },
      "children": Array [],
      "name": "button",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": null,
      "type": "tag",
      "x-attribsNamespace": Object {
        "aria-label": undefined,
        "class": undefined,
        "type": undefined,
      },
      "x-attribsPrefix": Object {
        "aria-label": undefined,
        "class": undefined,
        "type": undefined,
      },
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
